### PR TITLE
Custom CSS: prevent PHP warning

### PR DESCRIPTION
**Pull request by monsieur-z**
## Proposed changes:

On WordPress.com, the CSStidy library triggers a PHP warning: `Warning: Trying to access array offset on value of type`. This PR adds conditions in `class.csstidy-optimise.php` in the `jetpack-mu-wpcom` package to prevent the warning. This PR alone doesn't fix the issue for WordPress.com. We'll need to ensure it loads CSStidy from the `jetpack-mu-wpcom` package.

Note: the changes are reflected in the Jetpack plugin as well for good measure, though the Custom CSS will soon be removed from the plugin.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/362

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

This one's hard to test. I don't know what path triggered a warning. Adding tests seems overkill as well, given the scope of the changes and that the method's not pure.

Code review should be sufficient.

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/custom-css/csstidy/class.csstidy-optimise.php [hunk 1]
```diff
@@ -632,11 +632,11 @@ public static function explode_ws( $sep, $string ) {
 	 * @see dissolve_4value_shorthands()
 	 */
 	public static function merge_4value_shorthands( $array ) {
-		$return     = $array;
+		$return     = is_array( $array ) ? $array : array();
 		$shorthands = & $GLOBALS['csstidy']['shorthands'];
 
 		foreach ( $shorthands as $key => $value ) {
```
Comment (monsieur-z) [inline projects/packages/jetpack-mu-wpcom/src/features/custom-css/csstidy/class.csstidy-optimise.php:638]: An example of value for `$key => $value` is  `'padding' => [ 'padding-top', 'padding-right', 'padding-bottom', 'padding-left' ]`

File: projects/packages/jetpack-mu-wpcom/src/features/custom-css/csstidy/class.csstidy-optimise.php [hunk 2]
```diff
@@ -632,11 +632,11 @@ public static function explode_ws( $sep, $string ) {
...
-			if ( isset( $array[ $value[0] ] ) && isset( $array[ $value[1] ] )
+			if ( is_array( $value ) && isset( $array[ $value[0] ] ) && isset( $array[ $value[1] ] )
 							&& isset( $array[ $value[2] ] ) && isset( $array[ $value[3] ] ) && $value !== 0 ) {
```
Comment (darssen) [inline projects/packages/jetpack-mu-wpcom/src/features/custom-css/csstidy/class.csstidy-optimise.php]: Seems just moving the `$value !==0` check at the beginning might work due to the value in [data.inc.php](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/custom-css/csstidy/data.inc.php#L338) file of csstidy.

A similar thing was done [here](https://github.com/Cerdic/CSSTidy/commit/f75bbf97a9320b5b76b65a52b7962ded56e380ab) although the reason is not explicitly stated.

**Review by darssen — COMMENTED** [review]
Hi @monsieur-z, thanks for working on this.

Your solution should work, but I have added a comment on a solution that was used in the Upstream library. 

It took me a while to find it because the [current code](https://github.com/Cerdic/CSSTidy/blob/5810b0e7cc146c3c9a5075d0b005a836b78d14ae/data.inc.php#L271) does not have that property anymore. And also I don't really understand why they didn't use an empty array instead.

File: projects/packages/jetpack-mu-wpcom/src/features/custom-css/csstidy/class.csstidy-optimise.php (hunk 2 above)
Comment (monsieur-z) [reply projects/packages/jetpack-mu-wpcom/src/features/custom-css/csstidy/class.csstidy-optimise.php]: Thanks for the suggestion @darssen! Let's try that.

**Review by darssen — DISMISSED** [review]
LGTM!
Let's monitor if the Warnings disappear :)

**Review by darssen — APPROVED** [review]

**Pull request merged**